recipes-connectivity/networkmanager/networkmanager-1.4.4/xc-nutty-network.patch

```diff
@@ -0,0 +1,366 @@
++
++       if (rc.options->len) {
++               g_ptr_array_add (rc.options, NULL);
++               options = (char **) g_ptr_array_free (rc.options, FALSE);
++       } else
++               g_ptr_array_free (rc.options, TRUE);
++
++       nis_domain = rc.nis_domain;
++
++       strcpy(resolv_path, RESOLV_CONF);
++       strcat(resolv_path, ".");
++       strcat(resolv_path, iface);
++
++       if (success == FALSE)
++               success = update_resolv_conf (self, resolv_path, searches, nameservers, options, error, NM_DNS_MANAGER_RESOLV_CONF_MAN_FILE);
++
++       if (searches)
++               g_strfreev (searches);
++       if (nameservers)
++               g_strfreev (nameservers);
++       if (nis_servers)
++               g_strfreev (nis_servers);
++
++       return success;
++}
++
++static gboolean
+ nm_dns_manager_add_ip_config (NMDnsManager *self,
+                               const char *iface,
+                               gpointer config,
+@@ -1344,9 +1432,17 @@ nm_dns_manager_add_ip_config (NMDnsManag
+ 		g_clear_error (&error);
+ 	}
+ 
++	if (!xc_rewrite_resolv_conf (self, iface, config, FALSE, &error)) {
++			_LOGW ("Could not commit DNS changes: (%d) %s",
++				error ? error->code : -1,
++				error && error->message ? error->message : "(unknown)");
++		g_clear_error (&error);
++	}
++
+ 	return TRUE;
+ }
+ 
++
+ gboolean
+ nm_dns_manager_add_ip4_config (NMDnsManager *self,
+                                const char *iface,
+--- a/src/nm-policy.c
++++ b/src/nm-policy.c
+@@ -472,8 +472,10 @@ update_ip4_routing (NMPolicy *self, gboo
+ 	}
+ 	g_assert ((best || vpn) && best_ac);
+ 
++#if 0
+ 	if (!force_update && best && (best == priv->default_device4))
+ 		return;
++#endif
+ 
+ 	if (best) {
+ 		const GSList *connections, *iter;
+--- a/src/platform/nm-linux-platform.c
++++ b/src/platform/nm-linux-platform.c
+@@ -2284,6 +2284,7 @@ _nl_msg_new_route (int nlmsg_type,
+                    int nlmsg_flags,
+                    int family,
+                    int ifindex,
++                   int table,
+                    NMIPConfigSource source,
+                    unsigned char scope,
+                    gconstpointer network,
+@@ -2297,7 +2298,7 @@ _nl_msg_new_route (int nlmsg_type,
+ 	struct rtmsg rtmsg = {
+ 		.rtm_family = family,
+ 		.rtm_tos = 0,
+-		.rtm_table = RT_TABLE_MAIN, /* omit setting RTA_TABLE attribute */
++		.rtm_table = table,
+ 		.rtm_protocol = nmp_utils_ip_config_source_coerce_to_rtprot (source),
+ 		.rtm_scope = scope,
+ 		.rtm_type = RTN_UNICAST,
+@@ -5756,11 +5757,41 @@ ip4_route_add (NMPlatform *platform, int
+ {
+ 	NMPObject obj_id;
+ 	nm_auto_nlmsg struct nl_msg *nlmsg = NULL;
++	const char *iface = NULL;
++
++	iface = nm_platform_link_get_name(platform, ifindex);
++	if (iface) {
++		int table = 30 + ifindex;
++		guint32 oxt_metric = metric;
++		NMPObject oxt_obj_id;
++		nm_auto_nlmsg struct nl_msg *oxt_nlmsg = NULL;
++
++		if(strncmp(iface, "brbridged", 9) != 0)
++			metric = metric + 1;
```

Comment:
Not part of this PR: This should be configurable.

Reply:
Ack, I will remove it. Thanks

Reply:
Thanks, the delta between the metrics would not change the order anyway.
```
default via 192.168.0.1 dev brbridged proto static metric 425 
default via 192.168.0.1 dev wlan0 proto static metric 600 
```